docs/how-to/troubleshoot_deployment_issues.md

```diff
@@ -0,0 +1,184 @@
+# Troubleshoot the deployment issues
+
+This guide provides the steps to troubleshoot the unsuccessful Charmed Aether SD-Core deployments for different failure cases.
+
+## A. Create a Bug Report
+
+If you hit one of the following situations during the deployment, raise the issue by filling a [bug report][Bug Report].
+
+### A. 1. Terraform failed to deploy the Charmed Aether SD-Core because of configuration mismatch
+
+Terraform may fail to deploy the Charmed Aether SD-Core modules because of wrong configuration issues. These errors are generally caused by the mistakes in module designs or oversights during version changes.
+When the `terraform apply -auto-approve` command is run, deployment fails with wrong integration endpoints or wrong object attributes etc. as following:
+
+```console
+$ terraform apply -auto-approve 
+╷
+│ Error: Unsupported attribute
+│ 
+│   on main.tf line 54, in resource "juju_integration" "cu-nms":
+│   54:     endpoint = module.sdcore.fiveg_identity_endpoint
+│     ├────────────────
+│     │ module.sdcore is a object
+│ 
+│ This object does not have an attribute named "fiveg_identity_endpoint".
+```
+
+#### Recommended Actions
+
+[Validate the Terraform modules][Validate Terraform Configuration] by running `terraform validate` to analyze the dependencies between resources in your infrastructure configuration to determine the order to perform your operations.
+
+This command outputs the consistency result of the Terraform modules as success or error.
+
+Fixing the module may require advanced knowledge. If you hit a case that described above, create a [bug report][Bug Report]
```

Comment:
This is a how-to guide, please focus on specific steps and not so much on explanations. For instance here, you could simply say:

---

Validate the Terraform modules:

```shell
terraform validate
```

The command should not fail. If it does, please open a [bug report][Bug Report].

Reply:
Done